src/include/duckdb/execution/index/art/node4.hpp

```diff
@@ -14,50 +14,136 @@
 
 namespace duckdb {
 
-//! Node4 holds up to four Node children sorted by their key byte
+//! Node4 holds up to four children sorted by their key byte.
 class Node4 {
+	friend class Node16;
+
+public:
+	static constexpr NType NODE_4 = NType::NODE_4;
```

Comment:
Since the `Node4/Node16` and `Node7Leaf/Node15Leaf` share almost all of their functionality - maybe it would be cleaner to have a shared base class for these that they inherit from, e.g. something like:

```cpp
template<int CAPACITY, NodeType node_type>
class BaseNode {
...
};

class Node4 : public BaseNode<4, NodeType::NODE_4> {
   // ..
};

class Node16 : public BaseNode<16, NodeType::NODE_16> {
   // ..
};
```